packages/react/src/components/Search/Search-test.js

```diff
@@ -1,333 +1,157 @@
-          expect(icons.length).toBe(1);
-        });
-
-        it('renders two Icons', () => {
-          wrapper.setProps({ size: undefined });
-          const iconTypes = [SearchIcon, Close];
-          const icons = wrapper.findWhere((n) => iconTypes.includes(n.type()));
-          expect(icons.length).toEqual(2);
-        });
-      });
+      expect(container.firstChild).toHaveClass('custom-class');
     });
 
-    describe('Small Search', () => {
-      const small = mount(
-        <Search
-          id="test"
-          size="sm"
-          className="extra-class"
-          label="Search Field"
-          labelText="testlabel"
-        />
-      );
+    it('should respect closeButtonLabelText prop', () => {
+      render(<Search labelText="test-search" closeButtonLabelText="clear" />);
 
-      const smallContainer = small.find(`.${prefix}--search`);
+      expect(screen.getByLabelText('clear')).toBeInTheDocument();
+    });
 
-      it('renders correct search icon', () => {
-        const icons = small.find(SearchIcon);
-        expect(icons.length).toBe(1);
-      });
+    it('should respect defaultValue prop', () => {
+      render(<Search labelText="test-search" defaultValue="test-value" />);
 
-      it('should have the expected small class', () => {
-        expect(smallContainer.hasClass(`${prefix}--search--sm`)).toEqual(true);
-      });
+      expect(screen.getByRole('searchbox')).toHaveValue('test-value');
+    });
 
-      it('should only have 1 button (clear)', () => {
-        const btn = small.find('button');
-        expect(btn.length).toEqual(1);
-      });
+    it('should respect disabled prop', () => {
+      render(<Search labelText="test-search" disabled />);
 
-      it('renders two Icons', () => {
-        const iconTypes = [SearchIcon, Close];
-        const icons = wrapper.findWhere((n) => iconTypes.includes(n.type()));
-        expect(icons.length).toEqual(2);
-      });
+      expect(screen.getByRole('searchbox').disabled).toBe(true);
     });
-  });
-
-  describe('events', () => {
-    describe('onChange', () => {
-      const onChange = jest.fn();
 
-      beforeEach(() => {
-        onChange.mockReset();
-        render({ onChange: (e) => onChange(e.target.value) });
-      });
+    it('should respect id prop', () => {
+      render(<Search labelText="test-search" id="test-id" />);
 
-      describe('when input value is changed', () => {
-        const target = { value: 'test' };
-        const mock = { target };
+      expect(screen.getByRole('searchbox')).toHaveAttribute('id', 'test-id');
+    });
 
-        beforeEach(() => {
-          input().simulate('change', mock);
-        });
+    it('should respect labelText prop', () => {
+      render(<Search labelText="test-search" />);
 
-        it('is called', () => {
-          expect(onChange).toHaveBeenCalledWith(target.value);
-        });
-      });
+      expect(screen.getByRole('searchbox')).toBeInTheDocument();
+    });
```

Comment:
is testing that 'searchbox' is in the document sufficient to test for the labelText? 😅

Reply:
Zoinks! Great catch. I just updated it with a more thorough test